src/App.js

```diff
@@ -26,40 +26,92 @@ const generateSquares = () => {
 };
 
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+  const [playerTurn, setPlayerTurn] = useState(PLAYER_1);
+
+  const toggleTurn = () => {
+    setPlayerTurn(playerTurn === PLAYER_1 ? PLAYER_2 : PLAYER_1);
+  };
+
   const [squares, setSquares] = useState(generateSquares());
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const changeSquareValue = (id) => {
+    let newSquares = [];
+    for (let row of squares) {
+      let rowArray = [];
+      for (let column of row) {
+        if (column.id === id && !column.value) {
+          rowArray.push({id: column.id,
+          value: playerTurn});
+          toggleTurn();
+        } else 
+        {rowArray.push(column);}
+      } newSquares.push(rowArray); 
+    }
+    setSquares(newSquares);
+  };
 
   const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if
-    //    all three squares have the same value.
+    let squaresArray = [];
+
+    for (let rowIndex in squares) {
+      for (let objectIndex in squares[rowIndex]) {
+        squaresArray.push(squares[rowIndex][objectIndex].value);
+      }
+    }
+
+    // ROWS
```

Comment:
Nice work dividing the logic into rows, columns, and diagonals. Consider using loops to DRY up your code.